packages/lib/src/helpers/ast-transformer.ts

```diff
@@ -19,6 +33,197 @@ export const buildAstFromQuery = (queryStore: QueryItem[][]): AstTopLayer => {
+ * @returns array of different queries
+ */
+const divideQueries = (queries: AstTopLayer): QueryItem[][] => {
+    return queries.children.map((child) => divideCriterias(child));
+};
+
+/**
+ * divide a query into all different criteria
+ * NOTE: at the moment we don't expect a AstBottomLayer here, because we always have a logical AND between the criterias of a query
+ * @param query - a single query consiting of multiple criteria
+ * @returns a single query in the internal representation
+ */
+const divideCriterias = (query: AstElement): QueryItem[] => {
+    if (isTopLayer(query)) {
+        const criterias = query.children
+            .filter((child) => isTopLayer(child))
+            .map((child) => convertCriteria(<AstTopLayer>child));
+        return criterias != undefined ? <QueryItem[]>criterias : [];
+    } else {
+        return [];
+    }
+};
+
+/**
+ * convert a query criteria to it's internal query item representation
+ * NOTE: we have two cases here: 1) normal criteria represented as AstBottomLayerValue 2) complex criteria represented as an AstTopLayer
+ * @param criteria - a single query criteria
+ * @returns the internal QueryItem representation of this criteria
+ */
+const convertCriteria = (criteria: AstTopLayer): QueryItem | undefined => {
+    const values = criteria.children.filter((child) => isBottomLayer(child));
+    if (values.length === 0) return convertComplexCriteria(criteria);
+    const checkedValues: AstBottomLayerValue[] = <AstBottomLayerValue[]>values;
+    const convertedValues = convertValues(checkedValues);
+    const catalogueCategory = getCategoryFromKey(
+        get(catalogue),
+        checkedValues[0].key,
+    );
+    const convertedCriteria: QueryItem = {
+        id: uuidv4(),
+        key: checkedValues[0].key,
+        name: catalogueCategory != undefined ? catalogueCategory.name : "",
+        type: checkedValues[0].type,
+        system: checkedValues[0].system,
+        values: convertedValues,
+    };
+    if (catalogueCategory != undefined && "description" in catalogueCategory) {
+        convertedCriteria.description = catalogueCategory.description;
+    }
+    return convertedCriteria;
+};
+
+/**
+ * convert a query criteria based on an AstTopLayer to it's internal representation
+ * @param criteria - an AstTopLayer based criteria
+ * @returns the internal QueryItem representation of this criteria
+ */
+const convertComplexCriteria = (criteria: AstTopLayer): QueryItem => {
+    const values = criteria.children.filter((child) => isTopLayer(child));
+    const checkedValues: AstTopLayer[] = <AstTopLayer[]>values;
+    const convertedValues = convertValues(checkedValues);
+    const criteriaCateogory = getCategoryFromKey(get(catalogue), criteria.key);
+    return {
+        id: uuidv4(),
+        key: criteria.key,
+        name: criteriaCateogory != undefined ? criteriaCateogory.name : "",
+        // NOTE: Assuming hard coding here, proof me wrong
```

Comment:
For min/max we have between, see here https://github.com/samply/lens/blob/2b251f0bc14e490122eb19d17cba428b907b2001/packages/demo/public/catalogues/catalogue-dktk.json#L7117. Disproved!

Reply:
I also think we have somewhere a enum for all the types

Reply:
In this case we don't get a BETWEEN condition. The hard coding is only for the "complexCriterias" (multi-condition criterias, like dktk's tumor entities). In this case i think the top part should always be an "EQUALS" condition while all values below are parsed separatly as criteria.